app/Foundation/Providers/RepositoryServiceProvider.php

```diff
@@ -58,4 +62,23 @@ protected function registerMetricRepository()
             return new MetricRepository($repository, $dates);
         });
     }
+
+    /**
+     * Register the metric repository.
+     *
+     * @return void
+     */
+    protected function registerUpTimeRepository()
+    {
+        $this->app->singleton(UpTimeRepository::class, function (Container $app) {
+            $config = $app->make(ConfigRepository::class);
+
+            switch ($config->get('database.default')) {
+                case 'pgsql': $repository = new UpTimePgSqlRepository($config); break;
```

Comment:
What about SQLite?

Reply:
Sqlite doesn't support right joins apparently. I'm thinking that maybe a repository isn't the good solution. If this can be done with the eloquent ORM it would be better. The problem is a subquery.